### PR TITLE
Build based on overlays for each python version

### PR DESCRIPTION
**Pull request by harshad16**
Build based on overlays for each python version
Signed-off-by: Harshad Reddy Nalla <hnalla@redhat.com>

## Related Issues and Dependencies

Related-to: 
https://github.com/thoth-station/s2i-minimal-notebook/issues/36
https://github.com/thoth-station/s2i-minimal-notebook/issues/20

## Description

This will maintain dependencies on the overlays directory.

### CODE REVIEW

**Review by pacospace — APPROVED** [review]
lgtm!! 💯 👍🏼 🥇

**Comment by pacospace** [comment]
Fixes: https://github.com/thoth-station/jupyterlab-requirements/issues/53

**Comment by pacospace** [comment]
Related-To: https://github.com/thoth-station/thoth-application/issues/924

File: overlays/python38/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+FROM quay.io/thoth-station/s2i-thoth-ubi8-py38:v0.20.1
+ENV ENABLE_MICROPIPENV="1" \
```
Comment (goern) [inline overlays/python38/Dockerfile:2]: why is this not based on v0.24.2 ?

File: .aicoe-ci.yaml [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
-  registry: "quay.io"
-  registry-org: "thoth-station"
-  registry-project: "s2i-minimal-notebook"
-  registry-secret: "thoth-station-thoth-pusher-secret"
+  - overlays-build
+overlays_dir: overlays
+overlays:
+  - name: python36
+    build:
+      base-image: "quay.io/thoth-station/s2i-thoth-ubi8-py36:v0.20.1"
```
Comment (goern) [inline .aicoe-ci.yaml:7]: not v0.24.2 ?

**Review by goern — CHANGES_REQUESTED** [review]

**Comment by pacospace** [comment]
Related-To: https://github.com/opendatahub-io/odh-manifests/issues/327

**Comment by goern** [comment]
@harshad16 @pacospace is this good to be merged?!

**Comment by harshad16** [comment]
> is this good to be merged?!
not yet, i will update this PR soon.

**Pull request merged**